### PR TITLE
#170947692 user should be able to view most travelled destinations

### PR DESCRIPTION
**Pull request by EugeneMunya**
#### What does this PR do?
- Allow users to view most travelled  destinations
#### Description of Task to be completed?
- User can view the number of visits on each destination 
- display destinations order by number of visits the one with highest number of visit should come first
#### How should this be manually tested?
- open your terminal and run this command `git clone https://github.com/andela/knightsbn   backend.git` to clone repository
- run `cd knights-bn-backend` to get inside the project
- run `git fetch && git checkout ft-most-frequently-travelled-centre-170947692` 
- run `npm install` to install dependencies
- check if you have PostgreSQL database if you don't have it install it from here [PostgreSQL](https://www.2ndquadrant.com/en/blog/pginstaller-install-postgresql/)
- create database
- create `.env` file in your project root directory
- follow `env.example` file to create environment variables in `.env` file and 
   assign them with all your database configurations
- run `npm run dropTables` to clean your database
- run `npm run createTables` to create tables in your new database you have created.
- run `npm run dev-start` to start the server.
- open your postman or any other API tester you have
- you can log in as an admin with default credentials below
```
{
    "email": "traveladmin@gmail.com",
    "password": "Niyonkuru@1"
}
```
- create accommodations as an admin by making POST request to this URL `http://localhost:4000/api/v1/create/accommodation` along with the data bellow in the body
```
{
 "accommodationName":"Hill top",
 "locationName": "Chicago",
 "streetNumber": "104st ",
   "numberOfRooms": 4,
   "description":"it has wifi and big secreen TV",
    "availableRooms": [
      {
        "roomName": "Room 1",
        "roomType": "king",
        "price":"$50/night",
        "available":"true"
      },
       {
        "roomName": "Room 2",
        "roomType": "qween",
        "price":"$53/night",
        "available":"true"
      },
       {
        "roomName": "Room 3",
        "roomType": "qween",
        "price":"$54/night",
        "available":"true"
      }
      ,
       {
        "roomName": "Room 4",
        "roomType": "qween",
        "price":"$55/night",
        "available":"true"
      }
    ]
  }
```
- create the second accommodation as an admin using the data shown below
```
{
 "accommodationName":"Blue Sky",
 "locationName": "Nairobi",
 "streetNumber": "NM504st ",
   "numberOfRooms": 4,
   "description":"it has wifi and big secreen TV",
    "availableRooms": [
      {
        "roomName": "Room 1",
        "roomType": "king",
        "price":"$50/night",
        "available":"true"
      },
       {
        "roomName": "Room 2",
        "roomType": "qween",
        "price":"$53/night",
        "available":"true"
      },
       {
        "roomName": "Room 3",
        "roomType": "qween",
        "price":"$54/night",
        "available":"true"
      }
      ,
       {
        "roomName": "Room 4",
        "roomType": "qween",
        "price":"$55/night",
        "available":"true"
      }
    ]
  }
```
- login as requester using the default credentials below
```
{
    "email": "alain.maxime@gmail.com",
    "password": "Niyonkuru@1"
}
``` 
- send GET request  to this URL`  http://localhost:4000/api/v1/rooms/accommodations/3` to view available rooms
- book a room by sending a POST request to this URL `http://localhost:4000/api/v1/book/accommodations` along with the data below in the boy
```
{
"accomodationId": 3,
"roomName": "Room 1",
"checkinDate": "2020-06-13",
"checkoutDate": "2020-06-24"
}
```
- Log in again as another user by using the credentials below 
```
{
    "email": "ishimwewil005@gmail.com",
    "password": "Password@1"
}
```
- book an accommodation again using the data below
```
{
"accomodationId": 3,
"roomName": "Room 2",
"checkinDate": "2020-06-13",
"checkoutDate": "2020-06-24"
}
```
- login again as different user with this credentials below
```
{
    "email": "knights@gmail.com",
    "password": "Niyonkuru@1"
}
```
- book an accommodation using this data below
```
{
"accomodationId": 4,
"roomName": "Room 2",
"checkinDate": "2020-06-13",
"checkoutDate": "2020-06-24"
}
```
- to view most travelled destinations visit this URL  with GET request `http://localhost:4000/api/v1/most/traveled` and you will see how many visits for each destination has

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#170947692](https://www.pivotaltracker.com/story/show/170947692)
#### Screenshots
Screen Shot 2020-03-19 at 07 29 03

### CODE REVIEW

**Comment by T2Wil** [comment]
hey @EugeneMunya ,
you did a nice job but I think you may consider changing the keys'names of the most travelled places into more structured nouns: 
like instead of `Accommodation.accommodationName` put `accommodationName` and remember to use `camelCase` while naming variables : `numberofvisits` => `numberOfVisits`.
image

**Review by T2Wil — APPROVED** [review]
Nice work @EugeneMunya

**Review by Niyonsengaeric — APPROVED** [review]

File: src/controllers/accommodation.js [hunk 1]
```diff
@@ -113,5 +113,21 @@ export default class accomodationFacility {
       return res.status(500).json({ status: 500, errorMessage: error });
     }
   }
+
+  static async mostTraveled(req, res) {
+    const traveledCentre = await models.Bookings.findAll({
```
Comment (kagabof) [inline src/controllers/accommodation.js]: Why not using Booking from imports?

**Review by kagabof — CHANGES_REQUESTED** [review]

File: src/controllers/accommodation.js (hunk 1 above)
Comment (EugeneMunya) [reply src/controllers/accommodation.js]: I have switched to imported  you can check

File: src/controllers/accommodation.js [hunk 2]
```diff
@@ -113,5 +113,21 @@ export default class accomodationFacility {
       return res.status(500).json({ status: 500, errorMessage: error });
     }
   }
+
+  static async mostTraveled(req, res) {
+    const traveledCentre = await Bookings.findAll({
+      include: [
+        {
+          model: models.Accommodation,
```
Comment (kagabof) [inline src/controllers/accommodation.js]: Please use Accommodation from import.

**Review by kagabof — CHANGES_REQUESTED** [review]

File: src/controllers/accommodation.js (hunk 2 above)
Comment (EugeneMunya) [reply src/controllers/accommodation.js]: How I forgot this I thought I did it, let me do it and rebase also

**Pull request merged**